apis/sgv2-quarkus-common/src/main/java/io/stargate/sgv2/api/common/cql/builder/QueryBuilderImpl.java

```diff
@@ -1346,6 +1368,60 @@ private void appendConditions(
     }
   }
 
+  private void appendConditions(
+      Expression<BuiltCondition> whereExpression, String initialPrefix, StringBuilder builder) {
+    String prefix = initialPrefix;
+    if (initialPrefix.contains("IF") && ifExists) {
+      builder.append(prefix).append("EXISTS");
+      prefix = " AND ";
+    }
+    builder.append(initialPrefix); // must have where
+    builder.append(addExpressionCql(whereExpression));
+  }
+
+  private StringBuilder addExpressionCql(Expression<BuiltCondition> outerExpression) {
+    StringBuilder sb = new StringBuilder();
+    List<Expression<BuiltCondition>> innerExpressions = outerExpression.getChildren();
+    switch (outerExpression.getExprType()) {
+      case "and" -> {
+        sb.append(" ( ");
```

Comment:
Looks like extra spaces are still there; let's just remove from both side so

    sb.append("(");

and modify test accordingly. This is more along typical style of hand-written CQL queries.